server/test/services/withings/lib/withings.handler.test.js

```diff
@@ -0,0 +1,462 @@
+                    fm: 0,
+                    fw: 0,
+                  },
+                  {
+                    value: 0,
+                    type: 73,
+                    unit: 0,
+                    algo: 0,
+                    fm: 0,
+                    fw: 0,
+                  },
+                  {
+                    value: 0,
+                    type: 76,
+                    unit: 0,
+                    algo: 0,
+                    fm: 0,
+                    fw: 0,
+                  },
+                  {
+                    value: 0,
+                    type: 77,
+                    unit: 0,
+                    algo: 0,
+                    fm: 0,
+                    fw: 0,
+                  },
+                  {
+                    value: 0,
+                    type: 88,
+                    unit: 0,
+                    algo: 0,
+                    fm: 0,
+                    fw: 0,
+                  },
+                  {
+                    value: 0,
+                    type: 91,
+                    unit: 0,
+                    algo: 0,
+                    fm: 0,
+                    fw: 0,
+                  },
+                ],
+                comment: 'string',
+              },
+            ],
+            more: true,
+            offset: 0,
+          },
+        }),
+      },
+    });
+
+    server.start(done);
+
+    // Start fake oatuh2 server
+    // Generate a new RSA key and add it to the keystore
+    serverOauth2.issuer.keys.generateRSA();
+    // Start the server
+    serverOauth2.start(9292, 'localhost');
+  });
+
+  after(function testAfter(done) {
+    server.stop(done);
+    serverOauth2.stop();
+  });
+
+  const withingsHandler = new WithingsHandler(
+    gladys,
+    '55f177d7-bc35-4560-a1f0-4c58b9e9f2c4',
+    'http://localhost:9192',
+    'test',
+  );
+
+  it('withings serviceId is good', async () => {
+    const result = await withingsHandler.getServiceId();
+    return assert.equal(result.success, true);
+  });
+
+  it('withings setVar must send success = false', async () => {
+    const badGladys = {
+      variable: {
+        getValue: fake.returns(null),
+      },
+      event: new EventEmitter(),
+    };
+    const badWithingsHandler = new WithingsHandler(
+      badGladys,
+      '55f177d7-bc35-4560-a1f0-4c58b9e9f2c4',
+      'http://localhost:9192',
+      'test',
+    );
+
+    const result = await badWithingsHandler.saveVar('789dsfds452fsdq27fze2ds', 'fdsf847f5re3f92d1', 'test', null);
+    return assert.equal(result.success, false);
+  });
+
+  it('withings setVar must send success = true', async () => {
+    const result = await withingsHandler.saveVar('789dsfds452fsdq27fze2ds', 'fdsf847f5re3f92d1', 'test', null);
+    return assert.equal(result.success, true);
+  });
+
+  it('withings init/poll is good', async () => {
+    const resultInit = await withingsHandler.init('0cd30aef-9c4e-4a23-88e3-3547971296e5');
+    await assert.isNotNull(resultInit);
```

Comment:
The test is green even if the function do nothing...
![image](https://user-images.githubusercontent.com/32032645/105103339-bf96b100-5ab0-11eb-874c-000046d19dc9.png)

Reply:
I add a check in result